includes/admin/marketplace-suggestions/class-wc-marketplace-updater.php

```diff
@@ -0,0 +1,71 @@
+<?php
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Marketplace Suggestions Updater
+ */
+class WC_Marketplace_Updater {
+
+	public static function load() {
+		add_action( 'init', array( __CLASS__, 'init' ) );
+	}
+
+	/**
+	 * Schedule events and hook appropriate actions.
+	 */
+	public static function init() {
+		$queue = WC()->queue();
+		$next = $queue->get_next( 'woocommerce_update_marketplace_suggestions' );
+		if ( ! $next ) {
+			$queue->schedule_recurring( time(), WEEK_IN_SECONDS, 'woocommerce_update_marketplace_suggestions' );
+		}
+
+		add_action( 'woocommerce_update_marketplace_suggestions', array( __CLASS__, 'update_marketplace_suggestions' ) );
+	}
+
+	/**
+	 * Fetches new marketplace data, updates wc_marketplace_suggestions.
+	 */
+	public static function update_marketplace_suggestions() {
+		$data = get_option( 'woocommerce_marketplace_suggestions', array(
+			'suggestions' => array(),
+			'updated' => time(),
+		) );
+
+		$data['updated'] = time();
+
+		$url = 'https://d3t0oesq8995hv.cloudfront.net/add-ons/marketplace-suggestions.json';
+		$request = wp_remote_get( $url );
```

Comment:
Need to replace for `wp_safe_remote_get`

Reply:
It's a hard-coded URL which we already know is valid. Not sure why we would need to wp_http_validate_url() it again?

Reply:
Now using safe get: c45a39676c509546d6e44469324de34d8f621f2d